source/agora/node/API.d

```diff
@@ -198,4 +200,75 @@ public interface API
     ***************************************************************************/
 
     public Hash[] getMerklePath (ulong block_height, Hash hash);
+
+
+    /***************************************************************************
+
+        Get UTXOs from the UTXO set
+
+        API:
+            GET /get_utxos
+
+        Params:
+            address = used to identify the owner of UTXOs
+            type = transaction type of UTXOs to find
+            min_amount_value = mininum amount of UTXOs to find
+
+        Returns:
+            the associative array for UTXOs
+
+    ***************************************************************************/
+
+    public UTXOSetValue[Hash] getUTXOs (PublicKey address, TxType type,
```

Comment:
Why is this in the API ?? Same question for `createEnrollment` ?

Reply:
The getUTXOs can be used in the situation when a user(or Node client app) select an UTXO as a deposit for enrollment. So it shoud be open as API.
But, the createEnrollment could be inserted in the enrollValidator which must take an UTXO as parameter.

Reply:
`enrollVaidator` is an API endpoint receiving the enrollment data and cannot make enrollment data, and an user client should select an UTXO in the current scenario. So createEnrollment is needed now as an bridge between the two APIs. After the operations of an user Node client is determined, the createEnrollment API could be deleted.

Reply:
> The getUTXOs can be used in the situation when a user(or Node client app) select an UTXO as a deposit for enrollment. So it shoud be open as API.

Everything in `API` should be used *only* for interfacing with other nodes. There should not be *any* method here for convenience (by convenience, I mean used by non-node, e.g. APIs for exchanges).
Because such methods are easy DoS endpoint. Imagine a few clients calling this method if you have even [1% of the amount of UTXOs Bitcoin has today](https://www.blockchain.com/en/charts/utxo-count).

For convenience methods, we would expose *another* API, which is only enabled on demand. An API expected to be used only locally does not need to worry about this.

Reply:
> `enrollVaidator` is an API endpoint receiving the enrollment data and cannot make enrollment data, and an user client should select an UTXO in the current scenario. So `createEnrollment` is needed now as a bridge between the two APIs.

If you want to force a validator to enroll, just make an `enroll` method in [here](https://github.com/bpfkorea/agora/blob/1010eb23fdc9d702f5eea3e6dc4f30027da3dd73/source/agora/test/Base.d#L401-L415). This will not expose it to the world, but allow you to trigger specific actions on a node from an unittest.